src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1865,9 +1865,12 @@ private static Guid GenerateGuidFromName(string name)
                     {
                         dataType = Enum.GetUnderlyingType(dataType);
 
-                        int dataTypeSize = System.Runtime.InteropServices.Marshal.SizeOf(dataType);
-                        if (dataTypeSize < sizeof(int))
+                        if (dataType == typeof(byte) || dataType == typeof(sbyte)
+                            || dataType == typeof(bool) || dataType == typeof(char)
+                            || dataType == typeof(ushort) || dataType == typeof(short))
+                        {
                             dataType = typeof(int);
+                        }
```

Comment:
What you have is fine.  Might be a tad bit cleaner as a switch?
```C#
switch (Type.GetTypeCode(dataType))
{
    case TypeCode.Byte:
    case TypeCode.SByte:
    case TypeCode.Boolean:
    case TypeCode.Char:
    case TypeCode.Int16:
    case TypeCode.UInt16:
        dataType = typeof(int);
        break;
}
```

Reply:
Sure, fixed. It was par for the course if you look at the cascading ifs this is nested in, but looks better this way.